### PR TITLE
[Hackday][Messenger] Add an alias for transport.symfony_serializer so SerializerInterface can be autowired

### PR DESCRIPTION
**Pull request by karser**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes    
| License       | MIT

cc @thePanz 

Use case:
Before:
```
    Pnz\Messenger\FilesystemTransport\FilesystemTransportFactory:
        arguments:
            - '@messenger.transport.symfony_serializer'
        tags: ['messenger.transport_factory']
```

After:
```
    Pnz\Messenger\FilesystemTransport\FilesystemTransportFactory:
        tags: ['messenger.transport_factory']
```

### CODE REVIEW

**Comment by stof** [comment]
#SymfonyConHackday2018

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml [hunk 1]
```diff
@@ -22,7 +22,9 @@
             <argument /> <!-- Format -->
             <argument type="collection" /> <!-- Context -->
         </service>
-
+        <service id="Symfony\Component\Messenger\Transport\Serialization\SerializerInterface"
+                 alias="messenger.transport.symfony_serializer">
+        </service>
```
Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: We need a blank line here (for consistency and readability).

**Review by chalasr — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml (hunk 1 above)
Comment (karser) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: Done

**Review by chalasr — APPROVED** [review]
👍 with minor comment.
Sounds more like a feature to me, better merge it on master I'd say.

**Comment by ogizanagi** [comment]
I wondered about this alias before, but the serializer is configurable through the `framework.messenger.serializer.id` config. 
So what would you expect to be injected when setting this option to a different value than the `symfony_serializer` and typehinting Messenger's `SerializerInterface`?
Shouldn't it alias `messenger.transport.serializer` instead?

**Comment by chalasr** [comment]
Status: needs work
(https://github.com/symfony/symfony/pull/29517#discussion_r240009367)

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml [hunk 2]
```diff
@@ -22,6 +22,9 @@
             <argument /> <!-- Format -->
             <argument type="collection" /> <!-- Context -->
         </service>
+        <service id="Symfony\Component\Messenger\Transport\Serialization\SerializerInterface"
+                 alias="messenger.transport.symfony_serializer">
+        </service>
```
[the remark this answers is not in the document]
Comment (karser) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: That makes sense, done!

**Review by chalasr — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by karser** [comment]
@nicolas-grekas Did you mean this? please review

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml [hunk 3]
```diff
@@ -22,6 +22,8 @@
             <argument /> <!-- Format -->
             <argument type="collection" /> <!-- Context -->
         </service>
+        <service id="Symfony\Component\Messenger\Transport\Serialization\SerializerInterface"
+                 alias="messenger.transport.serializer" />
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: should be on one line

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @karser.

**Pull request merged**